src/network/terrier_server.cpp

```diff
@@ -38,44 +43,114 @@ TerrierServer::TerrierServer(common::ManagedPointer<ProtocolInterpreter::Provide
   signal(SIGPIPE, SIG_IGN);
 }
 
-void TerrierServer::RunServer() {
-  // This line is critical to performance for some reason
-  evthread_use_pthreads();
-
-  int conn_backlog = common::Settings::CONNECTION_BACKLOG;
+template<TerrierServer::SocketType type>
+void TerrierServer::RegisterSocket() {
+  static_assert(type == NETWORKED_SOCKET || type == UNIX_DOMAIN_SOCKET, "There should only be two socket types.");
+
+  constexpr auto conn_backlog = common::Settings::CONNECTION_BACKLOG;
+  constexpr auto is_networked_socket = type == NETWORKED_SOCKET;
+  constexpr std::string_view socket_description = is_networked_socket ? "networked" : "Unix domain";
+
+  auto &socket_fd = is_networked_socket ? network_socket_fd_ : unix_domain_socket_fd_;
+
+  // Gets the appropriate sockaddr for the given SocketType.
+  auto socket_addr = ([&] {
+    if constexpr (is_networked_socket) {
+      struct sockaddr_in sin = {0};
+
+      sin.sin_family = AF_INET;
+      sin.sin_addr.s_addr = INADDR_ANY;
+      sin.sin_port = htons(port_);
+
+      return sin;
+    } else {
+      // Builds the socket path name
+      // Naming conventions come from https://www.postgresql.org/docs/9.3/runtime-config-connection.html
+      const std::string socket_path = fmt::format("{0}/.s.PGSQL.{1}", socket_directory_, port_);
+      struct sockaddr_un sun = {0};
+      memset(&sun, 0, sizeof(struct sockaddr_un));
```

Comment:
What is the purpose of the memset here?

Reply:
Totally unneeded. Was based on what we used to do for network sockets, and I left that in by accident. Thanks for catching that!